xarray/core/indexing.py

```diff
@@ -1332,6 +1332,14 @@ def _indexing_array_and_key(self, key):
     def transpose(self, order):
         return self.array.transpose(order)
 
+    def oindex(self, key):
```

Comment:
As discussed, it might be easier to do one of these first (either `oindex` or `vindex`) but no strong preference on my end